### ISSUE TITLE
Notifications about negotiable resources

### ISSUE THREAD

**Issue by azaroth42** [IIIF/discovery#46]
The same resource URI might be an access point to both V2 and V3 resources, but only the V3 representation changes for a particular update.  Is it important that this be possible to express?
If so, how would we do it? And how would the crawler know to request V3 via content negotiation at all?

[Per Andy Irving at Edinburgh]

**Comment by aisaac** [comment]
[Discussed at Edinburgh too, with @azaroth42 ]
The problem of knowing which version to request seems a general issue/pattern of migration, not specific to the context of discovery.
And it is reasonable to assume that most (though not all) changes to a V3 manifest would also imply a change to the V2 manifest, from a semantic perspective. So it is rather safe to consider that a resource has changed when one of the representations has changed.

NB: A provider may maintain only their V3 manifests, but could be seen as a corner case...

**Comment by mixterj** [comment]
I tend to agree with @aisaac but there are some ramifications that we should discuss. For an Update event, the harvester could just look at both the V2 ad V3 Manifests (putting the burden on the harvester code to do that). The underlying issue for me is how to represent Collection Manifests between V2 and V3. In short can a V2 Collection Manifest contain nested V3 Manifests and visa versa. related to IIIF/api#1109

If you can not reference different version Manifests in a Collection Manifests, then you will get into a situation where a Collection could have a A/V item added to it, so an Update would occur but now the V2 and V3 Manifests are different. And worse if the harvester is just interested in V2 (and ignores V3 things) the Update notification would be a false positive because there would actually not be change to the V2 Collection Manifests

**Comment by mattmcgrattan** [comment]
Recommendation from Discovery call: 6th Feb 2019

This issue needs clarification, and then wider discussion within IIIF, as it's probably not a Discovery specific problem, and will have ramifications beyond Discovery.

**Comment by azaroth42** [comment]
Proposal for discovery:

In the discovery API, there would be one entry for each of the V2 and V3 URIs, rather than the negotiable resource. These would be the resources that are redirected to (in 3xx style negotiation) or in the `Content-Location` header (in 2xx style negotiation). They would likely both refer to the same metadata via `seeAlso` and could thus be aligned.

Then there's no ambiguity as to which version is used, but at the expense of being verbose about the changes.

```
    {
      "type": "Update",
      "object": {
        "id": "https://example.org/iiif/1/manifest/v2",
        "type": "Manifest",
        "seeAlso": "https://example.org/iiif/1/metadata.xml",
        "format": "application/ld+json;profile=\"https://iiif.io/api/presentation/2/context.json\""
      },
      "endTime": "2018-03-10T10:00:00Z"
    },
    {
      "type": "Update",
      "object": {
        "id": "https://example.org/iiif/1/manifest/v3",
        "type": "Manifest",
        "seeAlso": "https://example.org/iiif/1/metadata.xml",
        "format": "application/ld+json;profile=\"https://iiif.io/api/presentation/3/context.json\""
      },
      "endTime": "2018-03-10T10:00:00Z"
    }

```

**Comment by aisaac** [comment]
I like the pattern but am wondering whether we should really make it mandatory. 

It could be seen as best practice for publishers who want to keep legacy resources around. But not as a mandate for publishers to always publish separate URIs for v2 and v3 manifests. If they want to keep their life simple (possible at the expense of making their client's life more complex) then they can still keep the same URI, with content negotiation or not. With the provision that when the update happens is should be listed as such in their Discovery stream.

In a situation with content negotiation, then for the cases where the v2 resource doesn't change but the v3 resource changes, then this should be reflected by an update in the Discovery stream. That may not please v2 clients, but so be it. One could actually consider it's a good motivation for upgrade (i.e. the IIIF resource they're interested in has changed, possibly with some really cool stuff, but they can't tell what it is nor show it! Frustrating, isn't it? ;-) )

**Comment by azaroth42** [comment]
Discussion on call is +1 to proposal to be explicit about negotiated resources.

**Comment by aisaac** [comment]
Explicit in which sense? That our proposed pattern is different from the one with negotiated resources? That we think they should not be recommended? That we forbid them?

**Comment by azaroth42** [comment]
Call on 2019-03-20: Agree fixed, opened #48 to discuss link to canonical URI
